pkg/cli/clivalidation.go

```diff
@@ -161,10 +161,6 @@ func RequireApplicationArgs(cmd *cobra.Command, args []string, workspace workspa
 		return "", err
 	}
 
-	if applicationName == "" {
-		applicationName = workspace.DefaultApplication
-	}
-
 	if applicationName == "" {
 		applicationName = workspace.DirectoryConfig.Workspace.Application
 	}
```

Comment:
Why was the first if deleted and not the second one?

Reply:
As mentioned in the PR description we have two redundant ways for users to set the 'default application'. Line 164-167 is the per-user configuration file, which we're removing. Line 168-170 is the per-directory configuration file which will stay.